### PR TITLE
Tasks #202057 #202055 added filters for export facilitator list, and export ag list with subjects

### PR DESCRIPTION
**Pull request by yogesh-pawar-2023**
**Add this env variables:**
- `HASURA_SQL_BASE_URL=https://eg-dev.uniteframework.io/hasura/v2/query`
- `HASURA_DB_NAME=educategirlsdev` (Set according to the environments)

https://tracker.tekdi.net/issues/202057
https://tracker.tekdi.net/issues/202055?tab=time_entries

### CODE REVIEW

File: src/src/services/hasura/hasura.service.ts [hunk 1]
```diff
@@ -62,7 +62,49 @@ export class HasuraService {
 			console.log('get data error', e.message);
 		}
 	}
+	public async executeRawSql(sql: string) {
```
Comment (manojLondhe) [inline src/src/services/hasura/hasura.service.ts:65]: add new line before start of new method

File: src/src/services/hasura/hasura.service.ts [hunk 2]
```diff
@@ -62,7 +62,49 @@ export class HasuraService {
...
+			result = arr.slice(1).map((record) => {
+				const modifiedRecord = {};
+				columnNames.forEach((columnName, cNameIndex) => {
+					modifiedRecord[columnName] = record[cNameIndex];
+				});
+				return modifiedRecord;
+			});
+		}
+		return result;
+	}
```
Comment (manojLondhe) [inline src/src/services/hasura/hasura.service.ts:107]: add new line before start of new method

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 3]
```diff
@@ -172,6 +179,166 @@ export class BeneficiariesService {
 		}
 	}
 
+	async exportSubjectCsv(req: any, body: any, resp: any) {
+		try {
+			const user = await this.userService.ipUserInfo(req);
+			if (!user?.data?.program_users?.[0]?.organisation_id) {
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:185]: add blank lines before and after each `if`

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 4]
```diff
@@ -172,6 +179,166 @@ export class BeneficiariesService {
...
+                        created_at: ${sortType}
+                      }
+				) {
+					id
+					first_name
+					last_name
+					district
+					block
+					mobile
+				    program_beneficiaries {
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:248]: fix indent

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 5]
```diff
@@ -172,6 +179,166 @@ export class BeneficiariesService {
...
+			let result = response?.data?.users;
+			let mappedResponse = result;
+			console.log(
+				'mappedResponse',
+				mappedResponse[1],
+				mappedResponse.length,
+			);
+
+			const sql = `SELECT
+						name,
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:269]: fix indent

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 6]
```diff
@@ -172,6 +179,166 @@ export class BeneficiariesService {
...
+					enrolled_for_board
+					subjects
+					facilitator_id
+					status
+					
+				  }
+				}
+			  }`,
+			};
+			const response = await this.hasuraServiceFromServices.getData(data);
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:259]: add blank line after query

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 7]
```diff
@@ -172,6 +179,166 @@ export class BeneficiariesService {
...
+						`{program_beneficiaries:{status:{_eq:${status}}}}`,
+					);
+				}
+			}
+
+			let filterQuery = '{ _and: [' + filterQueryArray.join(',') + '] }';
+			var data = {
+				query: `query MyQuery {
+				users(where: ${filterQuery},
+                      order_by: {
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:238]: fix indents pls, make query easy to read

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 8]
```diff
@@ -172,6 +179,166 @@ export class BeneficiariesService {
...
+
+			const csvStringifier = createObjectCsvStringifier({
+				header: [
+					{ id: 'name', title: 'Name' },
+					{ id: 'enrolled_for_board', title: 'Enrolled For Board' },
+					...subjectHeader,
+				],
+			});
+			const records = [];
+			for (let data of mappedResponse) {
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:295]: add empty line before and after each `if`, `for`

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 9]
```diff
@@ -172,6 +179,166 @@ export class BeneficiariesService {
...
+						} else {
+							dataObject[allSubjects[i].name] = 'no';
+						}
+					} else {
+						dataObject[allSubjects[i].name] = 'no';
+					}
+				}
+				records.push(dataObject);
+			}
+			let fileName = `${
```
Comment (manojLondhe) [inline src/src/beneficiaries/beneficiaries.service.ts:321]: add comment

File: src/src/facilitator/facilitator.controller.ts [hunk 10]
```diff
@@ -97,9 +97,10 @@ export class FacilitatorController {
 
 	@Post('/exportCsv')
 	@UseGuards(new AuthGuard())
+	@UsePipes(ValidationPipe)
 	async exportFileToCsv(
 		@Req() request: any,
-		@Body() body: any,
+		@Body() body: FilterFacilitatorDto,
```
Comment (manojLondhe) [inline src/src/facilitator/facilitator.controller.ts:103]: where is this dto code? already committed?

File: src/src/facilitator/facilitator.service.ts [hunk 11]
```diff
@@ -1149,12 +1206,40 @@ export class FacilitatorService {
...
 			};
 			const hasuraResponse = await this.hasuraService.getData(data);
-			const allFacilitators = hasuraResponse?.data?.users;
+			let allFacilitators = hasuraResponse?.data?.users;
+			if (
+				allFacilitators &&
+				body.hasOwnProperty('work_experience') &&
+				this.isValidString(body.work_experience)
+			) {
+				const isValidNumberFilter =
```
Comment (manojLondhe) [inline src/src/facilitator/facilitator.service.ts:1232]: add comments for this logic?

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 3 above)
Comment (yogesh-pawar-2023) [reply src/src/beneficiaries/beneficiaries.service.ts:185]: added space

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 7 above)
Comment (yogesh-pawar-2023) [reply src/src/beneficiaries/beneficiaries.service.ts:238]: formatted

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 4 above)
Comment (yogesh-pawar-2023) [reply src/src/beneficiaries/beneficiaries.service.ts:248]: formatted

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 6 above)
Comment (yogesh-pawar-2023) [reply src/src/beneficiaries/beneficiaries.service.ts:259]: added

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 5 above)
Comment (yogesh-pawar-2023) [reply src/src/beneficiaries/beneficiaries.service.ts:269]: formatted

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 8 above)
Comment (yogesh-pawar-2023) [reply src/src/beneficiaries/beneficiaries.service.ts:295]: aded

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 9 above)
Comment (yogesh-pawar-2023) [reply src/src/beneficiaries/beneficiaries.service.ts:321]: added comments

File: src/src/facilitator/facilitator.controller.ts (hunk 10 above)
Comment (yogesh-pawar-2023) [reply src/src/facilitator/facilitator.controller.ts:103]: yes

File: src/src/facilitator/facilitator.service.ts (hunk 11 above)
Comment (yogesh-pawar-2023) [reply src/src/facilitator/facilitator.service.ts:1232]: added

File: src/src/services/hasura/hasura.service.ts (hunk 1 above)
Comment (yogesh-pawar-2023) [reply src/src/services/hasura/hasura.service.ts:65]: added as per comment

File: src/src/services/hasura/hasura.service.ts (hunk 2 above)
Comment (yogesh-pawar-2023) [reply src/src/services/hasura/hasura.service.ts:107]: added as per comment

**Pull request merged**